### PR TITLE
feat(elreal): Phase E.2 -- sqrt + hypot with depth-1 EFT refinement on sqrt

### PR DESCRIPTION
**Pull request by Ravenwater**
## Summary

Phase E.2 of epic #873. Implements [#888](https://github.com/stillwater-sc/universal/issues/888) (the second math sub-issue under Phase E #878).

Square root via Newton-style EFT residual at depth 1, plus a depth-0 hypot via `std::hypot` for overflow safety. **No dependency on Phase F's lazy division** -- the sqrt depth-1 correction divides by `2*c0` (a single double), not by a lazy elreal, so it works under the Phase C / Phase F constraint that lazy `/` is depth-0 only.

## `sqrt` algorithm

Newton's identity: `(c0 + delta)^2 = a  =>  delta = (a - c0^2) / (2*c0 + delta) ~= (a - c0^2) / (2*c0)` for `|delta| << c0`.

```text
c0       = std::sqrt(a.at(0))                  // depth 0
prod_err = two_prod_residual(c0, c0)           // c0^2 = prod_hi + prod_err exactly
num      = (a.at(0) - prod_hi) - prod_err + a.at(1)
c1       = num / (2 * c0)                       // depth 1; double divide, not lazy
```

So depth-1 captures both the EFT residual of `c0^2` and the operand's own depth-1 correction `a.at(1)`. Depth 2+ returns 0.0 (Heron iteration on the lazy stream is a Phase F follow-up).

Negative argument: throws `elreal_negative_sqrt_arg` under `ELREAL_THROW_ARITHMETIC_EXCEPTION`, else returns NaN via `std::sqrt` on the leading double.

## `hypot` algorithm

`std::hypot(a.at(0), b.at(0))` at depth 0. The standard library implementation handles overflow (large operands), signed zero, infinities, and NaN per IEEE-754 -- no need to re-derive the `max * sqrt(1 + (min/max)^2)` form here. Depth-1+ deferred; could be wired up later via `sqrt(a*a + b*b)` (when neither operand is near overflow) or the algebraic overflow-protected form walked at depth 1.

## Test plan

Three new binaries under `elastic/elreal/math/`:

| Binary | Coverage |
|---|---|
| `elreal_sqrt` | exact squares, non-square cross-validation vs `std::sqrt`, round-trip `sqrt(a)^2 ~= a`, `sqrt(2)` matches `std::numbers::sqrt2_v`, depth-1 propagation on rational input (`sqrt(2/3)`), negative -> NaN, special values |
| `elreal_sqrt_throw` | `elreal_negative_sqrt_arg` raised when `ELREAL_THROW_ARITHMETIC_EXCEPTION` is set; separate binary because the macro is fixed at compile time |
| `elreal_hypot` | Pythagorean triples (3-4-5, 5-12-13, 8-15-17), zero / symmetry / negative-arg cases, overflow safety at 1e200, cross-validation vs `std::hypot`, special-value propagation |

- [x] gcc 13: 16 binaries (3 new + 13 prior) all PASS
- [x] clang: 16 binaries all PASS

## What does *not* land in this PR

- Heron iteration / depth-2+ refinement for `sqrt` (Phase F)
- Depth-1+ refinement for `hypot` (small follow-up; the algebraic form needs care to stay overflow-safe at depth 1)

## Related

- Epic: #873
- Phase E parent: #878
- Phase E.2 issue: #888
- Phase E.1 (constants), merged via #893

## Summary by CodeRabbit

* **New Features**
  * Added sqrt() and hypot() for the elreal numeric type, including improved precision for sqrt (additional refinement) and robust handling of zeros, negatives, infinities, and NaNs; configurable behavior for arithmetic exceptions.

* **Tests**
  * New comprehensive test suites validating correctness, symmetry, overflow safety, round‑trip identities, leading‑zero/expansion edge cases, special‑value behavior, and throw vs non‑throw modes.

[Review Change Stack](https://app.coderabbit.ai/change-stack/stillwater-sc/universal/pull/894?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: include/sw/universal/number/elreal/elreal_impl.hpp [hunk 1]
```diff
@@ -876,6 +880,94 @@ inline elreal& elreal::operator/=(const elreal& rhs) { return *this = *this / rh
...
+#endif
+	}
+
+	double c0 = std::sqrt(a0);
+	elreal result;
+	result._components.push_back(c0);
+	result._computed_depth = 1;
+
+	// No depth-1 refinement makes sense for non-finite or zero leading.
+	if (!std::isfinite(c0) || c0 == 0.0) return result;
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply include/sw/universal/number/elreal/elreal_impl.hpp]: Fixed in 40a82899 -- real bug, well caught. The reproducer is the lazy-real distinctness case from Phase D: `elreal(1, 3) - elreal(1.0/3.0)` cancels at depth 0 and leaves a positive depth-1 correction equal to the rational residual. Pre-fix `sqrt` of that returned 0 (because `std::sqrt(a.at(0)) == 0`) instead of `~3.16e-5 = sqrt(1.85e-17)`. The negative case is just as bad: `{0.0, -1e-9}` represents a small negative value but `a.at(0) == 0` bypassed the negative-arg handler entirely.

Fix: walk materialised components to find the first non-zero. That component drives the sign check AND the leading `std::sqrt`. The depth-1 generator captures the index so the EFT residual formula uses the right leading and next-correction limbs. For the common case (lead_idx == 0) this collapses to the original formula.

Three new regression tests cover the fix:
- `sqrt(elreal(1, 3) - elreal(1.0/3.0))` round-trip recovers the input
- `sqrt({0, -1e-9})` produces NaN (not zero)
- `sqrt({0, +1e-9})` has leading ~`sqrt(1e-9) = 3.16e-5` (not zero)

Each of the three would have failed against the pre-fix code.
